### PR TITLE
OAuth with Github/Gitlab

### PR DESCRIPTION
**Pull request by mishal23**
# Status
Ready
**Note:** More custom permissions can be added as per need

# Description
- Adds support for GitHub & GitLab OAuth
- Moderation Panel for the Admin to delete the users in the Moderation Group and to change the group of a user.
- Currently supports two groups: Moderation & Moderated
- Can add more groups as per need.

# ToDos
- [x] Python 2&3 Support
- [x] PEP-8
- [x] Master branch up-to date
- [x] Tested Locally

# Steps to Test or Reproduce
Run the Django App locally, 
- create a new GitHub App at https://github.com/settings/developers, and update the credentials in `base.py`
- create a new GitLab App at https://gitlab.com/profile/applications, and update the credentials in `base.py`
- Create two groups named **Moderation** & **Moderated** from the Django Admin Panel, add permissions as per need.

# Deploy Notes
- Kindly add the necessary tokens before deploying.
- Kindly add permissions and groups before deploying.

### CODE REVIEW

File: src/accounts/templates/accounts/auth_base.html [hunk 1]
```diff
@@ -3,9 +3,8 @@
 {% load crispy_forms_tags %}
 
 {% block more_head %}
+<link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.9.0/css/all.css" integrity="sha384-i1LQnF23gykqWXg6jxC2ZbCbUMxyw5gLZY6UiUS98LYV5unm8GWmfkIS6jqJfb4E" crossorigin="anonymous">
```
Comment (abhijithanilkumar) [inline src/accounts/templates/accounts/auth_base.html]: Where is this key coming from?

File: src/accounts/views_admin.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+from profiles.models import Profile
+from django.http import HttpResponseRedirect
+
+User = get_user_model()
+
+
+@login_required
+def moderate_users(request):
+    users = User.objects.filter(groups__name='Moderation')
+    # query_set = Group.objects.filter(user = users)
```
Comment (abhijithanilkumar) [inline src/accounts/views_admin.py]: Remove the commented code if it is not used

File: src/accounts/views_admin.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+@login_required
+def delete_user(request):
+    if request.method == 'POST':
+        if 'delete' in request.POST and 'pk' in request.POST:
+            pk = request.POST['pk']
+            try:
+                user = User.objects.get(pk=pk)
+            except Exception:
+                return
```
Comment (abhijithanilkumar) [inline src/accounts/views_admin.py]: Shall we return and print a message to the front end? This looks like an abrupt end.

File: src/templates/base2.html [hunk 4]
```diff
@@ -119,7 +122,7 @@ <h1>New Project</h1>
     <!-- Bootstrap core JavaScript
     ================================================== -->
     <!-- Placed at the end of the document so the pages load faster -->
-    <script src="https://code.jquery.com/jquery-1.10.2.min.js"></script>
+    <script src="//code.jquery.com/jquery-1.10.2.min.js"></script>
```
Comment (abhijithanilkumar) [inline src/templates/base2.html]: Is this change required?

File: src/accounts/views_admin.py (hunk 3 above)
Comment (mishal23) [reply src/accounts/views_admin.py]: Actually, it'll not reach anytime there, as we don't show the users name on the list if the user does not exist.

File: src/accounts/templates/accounts/auth_base.html (hunk 1 above)
Comment (mishal23) [reply src/accounts/templates/accounts/auth_base.html]: It isn't necessary, removed :+1:
